File: solutions/zig/isogram/1/isogram.zig
const std = @import("std");

pub fn isIsogram(str: []const u8) bool {
    var letters: [26]bool = .{false} ** 26;
    for (str) |c| {
        if (std.ascii.isAlphabetic(c)) {
            if (letters[std.ascii.toLower(c) - 'a']) return false;
            letters[std.ascii.toLower(c) - 'a'] = true;
        }
    }

    return true;
}

File: solutions/c/isogram/1/isogram.c
#include "isogram.h"
#include <ctype.h>
#include <string.h>

bool is_isogram(const char phrase[]) {
  int length, count;
  if (phrase == NULL)
    return false;

  length = strlen(phrase);
  for (int i = 0; i < length; i++) {
    count = 0;
    for (int j = 0; j < length; j++) {
      if (phrase[i] == ' ' || phrase[i] == '-')
        continue;
      if (tolower(phrase[i]) == tolower(phrase[j]))
        count++;
    }
    if (count > 1)
      return false;
  }

  return true;
}
